packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2254,29 +2254,30 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     // current composing region) is very infinite-loop-prone: the formatters
     // will keep trying to modify the composing region while Gboard will keep
     // trying to restore the original composing region.
-    final bool textChanged = _value.text != value.text
-                          || (!_value.composing.isCollapsed && value.composing.isCollapsed);
-    final bool selectionChanged = _value.selection != value.selection;
+    final bool preformattedTextChanged = _value.text != newTextEditingValue.text
+                                      || (!_value.composing.isCollapsed && newTextEditingValue.composing.isCollapsed);
```

Comment:
changed the parameter name because it was making this expression kinda confusing.